app/src/main/java/com/rockthevote/grommet/ui/registration/review/ReviewAndConfirmFragment.java

```diff
@@ -162,6 +185,7 @@ public void onRegisterClick(View v) {
             dialog.show(getFragmentManager(), "complete_dialog");
 
             LocaleUtils.setLocale(new Locale("en"));
+            storeState();
```

Comment:
I don't think we need to update the view model here.

There's no override implementation of `verify()`in this fragment. Once "register" is clicked, it just makes sure there's something (anything) in the signature box and then opens the confirm dialog fragment, which originally would trigger the network call.

In this case, I think once "register" is clicked, we can just include a call to "storeState" right before we show the dialog fragment.

Reply:
> In this case, I think once "register" is clicked, we can just include a call to "storeState" right before we show the dialog fragment.

Isn't that what this is doing? Not in exact line order, but two lines above this is the call to show the dialog.

Alternatively, rather than `storeState` just handling storing the state to the ViewModel, we could have it be responsible for building and databasing the JSON blob for the request. I'm still debating if it makes sense to do this here for simplicity, or turn that control over to the Activity for more proper state management.

Reply:
>Isn't that what this is doing? 
damn, you're right, didn't notice the collapsed lines